pkg/engine/hybrid/crawl.go

```diff
@@ -78,13 +75,13 @@ func (c *Crawler) navigateRequest(ctx context.Context, httpclient *retryablehttp
 		rawBytesResponse, _ := httputil.DumpResponse(httpresp, true)
```

Comment:
@Mzack9999 , suggesting use of retryablehttp.Request here and [here](https://github.com/projectdiscovery/katana/blob/f38f5a0d7bf62861a70eeaec3940071c544abbff/pkg/engine/hybrid/crawl.go#L61-L62)
- we recently added `.Dump` method to dump request and response 
- context in `s` can be used with `retryablehttp.NewRequest` and we  should defer it avoid memory leaks